src/claimtrie.cpp

```diff
@@ -2554,10 +2557,75 @@ uint256 CClaimTrieCache::getLeafHashForProof(const std::string& currentPosition,
     }
 }
 
-CClaimTrieProof CClaimTrieCache::getProofForName(const std::string& name) const
+void CClaimTrieCache::recursiveFlatten(const std::string& name, const CClaimTrieNode* current, std::vector<namedNodeType>& nodes) const
+{
+    nodes.push_back(std::make_pair(name, *current));
+    nodeCacheType::const_iterator cachedNode;
+    for (nodeMapType::const_iterator it = current->children.begin(); it != current->children.end(); ++it)
+    {
+        std::string str = name + char(it->first);
+        cachedNode = cache.find(str);
+        if (cachedNode != cache.end())
+            recursiveFlatten(str, cachedNode->second, nodes);
+        else
+            recursiveFlatten(str, it->second, nodes);
+    }
+}
+
+
+std::vector<namedNodeType> CClaimTrieCache::flattenTrie() const
```

Comment:
This method existed for CClaimTrie but not for the "cache".

Reply:
What is this supposed to do?  It doesn't make sense to me.

Reply:
We need ability read cached claim, it makes sense when we use rollback feature.